geomstats/special_orthogonal_group.py

```diff
@@ -135,6 +135,72 @@ def regularize(self, rot_vec):
         assert regularized_rot_vec.ndim == 2
         return regularized_rot_vec
 
+    def regularize_tangent_vec_at_identity(self, tangent_vec, metric=None):
+        if metric is None:
+            metric = self.left_canonical_metric
+        if tangent_vec.ndim == 1:
+            tangent_vec = np.expand_dims(tangent_vec, axis=0)
+
+        tangent_vec_metric_norm = metric.norm(tangent_vec)
+        tangent_vec_canonical_norm = np.linalg.norm(tangent_vec, axis=1)
+        if tangent_vec_canonical_norm.ndim == 1:
+            tangent_vec_canonical_norm = np.expand_dims(
+                                     tangent_vec_canonical_norm, axis=1)
+
+        mask_norm_0 = np.isclose(tangent_vec_metric_norm, 0)
+        mask_canonical_norm_0 = np.isclose(tangent_vec_canonical_norm, 0)
+
+        mask_0 = mask_norm_0 | mask_canonical_norm_0
+        mask_else = ~mask_0
+
+        mask_0 = np.squeeze(mask_0, axis=1)
+        mask_else = np.squeeze(mask_else, axis=1)
+
+        coef = np.empty_like(tangent_vec_metric_norm)
+        regularized_vec = tangent_vec
+        if np.any(mask_0):
```

Comment:
Remind me why you need the np.any tests here? Shouldn’t this work directly(meaning if there are no elements in mask 0 it won’t idnex anything so you get less lines of code?)